### DISCUSSION TITLE
Allow notebook filters (ipynb-filters) as Quarto Extensions

### DISCUSSION THREAD

**Discussion by roberttorfason** [quarto-dev/quarto-cli#5894] [Feature Requests]
### Description

Is there a way to distribute notebook filters on a project level in the same/similar way as lua filters i.e. using `quarto add my_notebook_filter`?

What I would like to achive is to create an extension that contributes a `ipynb-filters` e.g.

```
title: My Notebook Filter
author: Me
version: 0.1.0
quarto-required: ">=1.2.198"
contributes:
  ipynb-filters:
    - my_notebook_filter.py
```

However if I try this it complains with the error `ERROR: The extension My Notebook Filter is not valid- it does not contribute anything.`

My current workaround is to just add a dummy lua filter to be able to install it and then refer to the filter using
```
ipynb-filters:
  - _extensions/my_notebook_filter/my_notebook_filter.py
```

But it would be great to be able to just refer to it by name in the same way you can do with lua filters e.g.
```
ipynb-filters:
  - my_notebook_filter
```

However as far as I can currently tell, the only way to distribute/reuse notebook filters is to copy paste the files to the project directory.

Is there a way to do this? If not, and there is interest I would be happy to take a shot at contributing to it.

**Comment by cderv** [comment]
@dragonstyle you added the extensions features. Do we do a allow extensions for notebook filters ? Is this not possible on purpose ?

**Reply by dragonstyle** [reply]
For anything that is extensible via extensions, we do need to implement support (e.g. when resolving nbfilters, we need to have the concept of extensions contributing nbfilters and check whether any do). I haven't implemented that, though it would be a nice enhancement request! It is a pretty small amount of work for sure...

**Reply by cderv** [reply]
Let's move this to a feature request then ! Thanks.
